### PR TITLE
GH-41119: [Archery][Packaging][CI] Avoid using --progress flag on Docker on Windows on archery

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

Windows wheels are currently failing due to the change on `ARCHERY_DEBUG=1` by default. This uses `--progress` on `docker build` which is not supported on Windows.

### What changes are included in this PR?

Do not use `--progress` on the Windows builds.

### Are these changes tested?

Yes on CI via archery.

### Are there any user-facing changes?
No
* GitHub Issue: #41119

### CODE REVIEW

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-windows-*

**Comment by raulcd** [comment]
@kou are you ok with this? This is needed for 16.0.0

File: .env [hunk 1]
```diff
@@ -98,7 +98,7 @@ VCPKG="a42af01b72c28a8e1d7b48107b33e4f286a55ef6"    # 2023.11.20 Release
 # ci/docker/python-wheel-windows-vs2019.dockerfile.
 # This is a workaround for our CI problem that "archery docker build" doesn't
 # use pulled built images in dev/tasks/python-wheels/github.windows.yml.
-PYTHON_WHEEL_WINDOWS_IMAGE_REVISION=2024-03-19
+PYTHON_WHEEL_WINDOWS_IMAGE_REVISION=2024-04-10
```
Comment (kou) [inline .env:101]: Can we revert this?

**Review by kou — APPROVED** [review]
+1

**Comment by kou** [comment]
Or can we remove --progress on Windows?

File: .env (hunk 1 above)
Comment (raulcd) [reply .env]: Done

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-windows-*

**Comment by raulcd** [comment]
> Or can we remove --progress on Windows?

@kou I've updated the PR to do that.

**Comment by kou** [comment]
@github-actions crossbow submit wheel-windows-*

**Review by kou — APPROVED** [review]
+1

No `--progress=plain`:

https://github.com/ursacomputing/crossbow/actions/runs/8637649581/job/23680319104#step:7:72

```text
DEBUG:archery:Executing `['docker', 'build', '--build-arg', 'BUILDKIT_INLINE_CACHE=1', '--build-arg', 'python=3.11', '--build-arg', 'vcpkg=a42af01b72c28a8e1d7b48107b33e4f286a55ef6', '-f', 'D:\\a\\crossbow\\crossbow\\arrow\\ci/docker/python-wheel-windows-vs2019.dockerfile', '-t', 'ghcr.io/ursacomputing/arrow:python-3.11-wheel-windows-vs2019-vcpkg-a42af01b72c28a8e1d7b48107b33e4f286a55ef6-2024-04-11', 'D:\\a\\crossbow\\crossbow\\arrow']`
```

**Pull request merged**